vars/piperPipelineStageSecurity.groovy

```diff
@@ -32,17 +36,64 @@ void call(Map parameters = [:]) {
         .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
         .mixinStageConfig(script.commonPipelineEnvironment, stageName, STEP_CONFIG_KEYS)
         .mixin(parameters, PARAMETER_KEYS)
+        .addIfEmpty('checkmarxExecuteScan', script.commonPipelineEnvironment.configuration.runStep?.get(stageName)?.checkmarxExecuteScan)
+        .addIfEmpty('fortifyExecuteScan', script.commonPipelineEnvironment.configuration.runStep?.get(stageName)?.fortifyExecuteScan)
         .addIfEmpty('whitesourceExecuteScan', script.commonPipelineEnvironment.configuration.runStep?.get(stageName)?.whitesourceExecuteScan)
         .use()
 
     piperStageWrapper (script: script, stageName: stageName) {
 
-        // telemetry reporting
-        utils.pushToSWA([step: STEP_NAME], config)
+        def securityScanMap = [:]
+
+        if (config.checkmarxExecuteScan) {
+            securityScanMap['Checkmarx'] = {
+                node(config.nodeLabel) {
+                    try{
+                        durationMeasure(script: script, measurementName: 'checkmarx_duration') {
+                            checkmarxExecuteScan script: script
+                        }
+                    }finally{
+                        deleteDir()
```

Comment:
Will it work in all cases to deleteDir() here if we have multiple steps in the stage? Is the parallel step guaranteed to execute the steps in different nodes? (Sorry if this exposes my ignorance of Jenkins workings...)

Reply:
since we define a dedicated node per parallel stage Jenkins will ensure this.
We have been using it like that for quite a long time and I am not aware of any side effect.